### ISSUE TITLE
Broken Rendering when applying Neos.Neos:ContentCollection as prop to to Neos.Fusion:Component renderer

### ISSUE THREAD

**Issue by bweinzierl** [neos/neos-development-collection#3003]
### Description

After Upgrading from Neos 5.1 to 5.2 rendering is broken for every content element where a Neos.Neos:ContentCollection is used and applied as prop.

### Steps to Reproduce

1. Create the following content element with a ContentCollection as childNode:

```
prototype(My.Package:Content.Section) < prototype(Neos.Neos:ContentComponent) {
	test = ${q(node).property('title')}
	content = Neos.Neos:ContentCollection {
		nodePath = 'content'
	}
	renderer = Neos.Fusion:Component {
		@apply.props = ${props}
		renderer = afx`
			<div>{props.content}</div>
		`
	}
}
```

2. The following rendering error appears in frontend when ContentElement used on a page

```
An exception was thrown while Neos tried to render your page
No Fusion object found in path "root<Neos.Fusion:Case>/documentType<Neos.Fusion:Matcher>/element<My.Website:Document.MetaPage>/body<Neos.Fusion:Component>/content<Neos.Fusion:Join>/main<Neos.Neos:PrimaryContent>/default<Neos.Fusion:Matcher>/renderer<Neos.Neos:ContentCollection>/content<Neos.Neos:ContentCollectionRenderer>/itemRenderer<Neos.Neos:ContentCase>/default<Neos.Fusion:Matcher>/element<My.Package:Content.Section>/content<Neos.Neos:ContentCollection>/content<Neos.Neos:ContentCollectionRenderer>/itemRenderer<Neos.Neos:ContentCase>/test" Please make sure to define one in your Fusion configuration.
```

#### Expected behavior

This should display the content of the contentCollection wrapped by a div

#### Actual behavior

Rendering error is shown

### Affected Versions

On Neos 5.2 but not on 5.1

### Additional information

Discussion in Slack: https://neos-project.slack.com/archives/C050C8FEK/p1588946859478700
Might be related to performance optimisations

**Comment by Sebobo** [comment]
Can reproduce as discussed.

Maybe @mficzel or @hlubek have an idea.

**Comment by Sebobo** [comment]
I encountered a similar error yesterday when using `@context` in the component outside the renderer without using `@apply`. Will see if I can build a small test case.

**Comment by mficzel** [comment]
Just a quick hint. I am pressty sure this is a problem with https://github.com/neos/neos-development-collection/pull/2738 

@bweinzierl can you verify wether this is caused by the wrapper component pattern
```
renderer = Neos.Fusion:Component {
    @apply.props = ${props}
```
or by passing the ContentCollection as prop or by the combination of both.

**Comment by hlubek** [comment]
Hi! Yes, #2738 could cause this. Would be good to write the simplest possible functional test to reproduce it. I can help with the debugging / fix.

**Comment by Sebobo** [comment]
Tried writing a functional test for this yesterday but wasn't able yet only by using Neos.Fusion objects. 
Don't fully understand which part somehow moves test into the ContentCase

**Comment by bweinzierl** [comment]
> @bweinzierl can you verify wether this is caused by the wrapper component pattern or by passing the ContentCollection as prop or by the combination of both.

Is is the combination of both. 

This Works (collection is not used in renderer):
```
	test = ${q(node).property('title') + 'sdfsd'}

	content = Neos.Neos:ContentCollection {
		nodePath = 'content'
	}

	renderer = Neos.Fusion:Component {
		@apply.props = ${props}

		renderer = afx`
			<div clas="adfasd">{props.test}</div>
		`
	}
```

This also works (no wrapper component but collection is used in renderer):
```
	test = ${q(node).property('title') + 'sdfsd'}

	content = Neos.Neos:ContentCollection {
		nodePath = 'content'
	}

	renderer = afx`
		<div clas="adfasd">{props.content}</div>
	`
```

**Comment by klfman** [comment]
Can confirm this problem as well. Is there any known workaround to not depend on a downgrade to Neos 5.1?

**Comment by KommunikativCh** [comment]
Hello I have found the situation which causes this.
In Neos.Fusion/Classes/Core/Runtime.php  on line 677 __objectType ist now loading with Neos.Fusion:Lazy
In CaseImplementation.php on line 69 this __objectType is checking of existence before this was not set now it ist ‘Neos.Fusion;Lazy’ and calls the runtime->render without Neos.Fusion:Matcher

**Comment by mficzel** [comment]
The best way to proceed here would be to provide a functional test that fails because of this issue. I am pretty shure that once we have this the fix will be fast.

**Comment by mficzel** [comment]
Until we have a test  and a solution maybe the safe way would be to revert the lazy props change for now?

**Comment by hlubek** [comment]
> Until we have a test and a solution maybe the safe way would be to revert the lazy props change for now?

With the feedback of @KommunikativCh I'll try to implement a functional test, so let's wait for a day or two. Maybe we can solve this issue with a bugfix release.

**Comment by hlubek** [comment]
I found the minimal functional test case and also the buggy part in `Runtime`:

```
component.wrapperRenderer = Neos.Fusion:Component {
  title = 'Hello'

  content = Neos.Fusion:Component {
    renderer = Neos.Fusion:Case {
      default {
        @position = 'end'
        condition = ${true}
        renderer = 'Default content'
      }
    }
  }

  renderer = Neos.Fusion:Component {
    @apply.props = ${props}

    renderer = ${props.content}
  }
}
```

The issue here is, that apply values are _always_ set on all nested Fusion objects (if they extend `AbstractArrayFusionObject`) as the "storage" of actual apply values was changed with 0642115d529237bd30ae1502a78409e7739c4718 to not use a stack, but rather a lookup per (absolute) Fusion path to allow lazy evaluation with out of stack order. But the code in `\Neos\Fusion\Core\Runtime::setPropertiesOnFusionObject` wasn't changed accordingly (which is quite funny that it worked until this scenario came around) - so it just applied all the remembered values for paths from upper level `@apply`. 🤷‍♂️ 

I'll create a patch with an according fix for the Runtime.
